samtranslator/swagger/swagger.py

```diff
@@ -50,6 +52,28 @@ def has_path(self, path, method=None):
 
         return result
 
+    def get_integration_from_method(self, method):
+        """
+        Returns the integration for this method.
+
+        :param dict method: method dictionary
+        :return: X_APIGW_INTEGRATION section for this method
+        """
+        return self.get_method_contents(method).get(self._X_APIGW_INTEGRATION)
+
+    def get_method_contents(self, method):
+        """
+        Returns the swagger contents of the given method. This checks to see if a conditional block
+        has been used inside of the method, and, if so, returns the method contents that are
+        inside of the conditional.
+
+        :param dict method: method dictionary
+        :return: swagger components of the method
+        """
+        if self._CONDITIONAL_IF in method:
+            return method[self._CONDITIONAL_IF][1]
```

Comment:
This assumes the method is in this position, but this is not always true. For example, I could have a condition that when evaluated to true returns NoValue and when false returns the method.

Reply:
Good point. SAM doesn't set something like that up, but someone writing Swagger could.

Reply:
This is a very good callout. Will have to think about this.

Reply:
Ok. Couple of thoughts here.

This has never worked up to this point, which means that no existing SAM templates will have a conditional inside of a method configuration. So, whatever we do here won't *break* any existing customers, but could potentially affect *future* uses of SAM.

SAM uses this code, I think, only for two things:
1. Creating Swagger for the user (generating APIGW paths and methods)
1. Adding an authorizer to a method.

This means that the only time (at least as SAM is now) a customer would have SAM read their swagger is when they are using custom authorizers. We should still assume that other use cases could arise.

Possible solutions:

1. Leave as is:
    1. Document that SAM only modifies the first element of a Condition that is nested under a method.
    1. This means that, with authorizers, SAM would only add an authorizer to the first item in the conditional and ignore the second entirely. 
    1. SAM will also throw an error if the first item is invalid.
1. Check the first item to see if it is a valid path, and, if not, return the second item.
    1. This is an either/or situation. It still wouldn't handle the case where both paths would need to be modified.
1. Just do whatever we're doing to both items in the list, and throw an error only if both of them are invalid paths.
    1. Might be the safest approach.  
    1. Would a customer ever want SAM to only modify the first one and not touch the second? 
    1. Or should we return both parts of this conditional list and make sure that it is a valid path before modifying it (for authorizers or other similar things we do in the future)?

Reply:
The third option might be the best option here, should we go ahead and modify all valid method configurations that are in the conditional list? The change is pretty small I think

Reply:
3rd option sounds the best